src/ray/raylet/cgroup_v2_utils.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/raylet/cgroup_v2_utils.h"
+
+#include <fstream>
+#include <string>
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/strip.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+bool IsCgroupV2MountedAsRw() {
```

Comment:
nits: I think you also need to check that the normal user in the container can modify the cgroup path, usually `/sys/fs/cgroup` is owned by `root` user even though it's writable, it can only writable by the `root`.

See https://github.com/anyscale/product/pull/32052/files#diff-005124dd6eac41dac5da8cac101b497921e0ee33ca99ec034fbb9fd82ddba508

Reply:
I think it's a good point, to update cgroupv2, mounted as rw is not enough, current user should also have the permission to write. Add a test to check.

Reply:
Another thought, another way we could do is we don't do any precheck before actually setting cgroup, which fails directly due to lacking permission.